test/screen/HtmlScreen.js

```diff
@@ -4,24 +4,22 @@ import dom from 'metal-dom';
 import globals from '../../src/globals/globals';
 import HtmlScreen from '../../src/screen/HtmlScreen';
 import Surface from '../../src/surface/Surface';
-import UA from 'metal-useragent';
+import utils from '../../src/utils/utils';
 import Uri from 'metal-uri';
 
 describe('HtmlScreen', function() {
 
+	let fetchStub;
+
 	beforeEach(() => {
-		var requests = this.requests = [];
-		this.xhr = sinon.useFakeXMLHttpRequest();
-		this.xhr.onCreate = (xhr) => {
-			requests.push(xhr);
-		};
 		// Prevent log messages from showing up in test output.
 		sinon.stub(console, 'log');
+		fetchStub = sinon.stub(window, 'fetch');
 	});
 
 	afterEach(() => {
-		this.xhr.restore();
 		console.log.restore();
+		fetchStub.restore();
```

Comment:
Any reason why the pattern used for `console.log` wouldn't work here in the same way? ie.

```javascript
sinon.stub(window, 'fetch'); // before

window.fetch.restore(); //after
```

(Presuming that you can later on call `window.fetch.returns(...)` etc. I don't know if that works or not or whether you need to save a reference to the value returned by `sinon.stub()`)

Reply:
Thanks. Using the same pattern is a much better idea. Using `window.fetch.returns(...)` does exactly what we want so we don't need to save a reference to the value returned by `sinon.stub()`